samtranslator/model/preferences/deployment_preference_collection.py

```diff
@@ -152,6 +152,69 @@ def deployment_group(self, function_logical_id):
 
         return deployment_group
 
+    def _convert_alarms(self, preference_alarms):
+        """
+        Converts deployment preference alarms to an AlarmsConfiguration
+
+        Parameters
+        ----------
+        preference_alarms : dict
+            Deployment preference alarms
+
+        Returns
+        -------
+        dict
+            AlarmsConfiguration if alarms is set, None otherwise
+
+        Raises
+        ------
+        ValueError
+            If Alarms is in the wrong format
+        """
+        if not preference_alarms or is_intrinsic_no_value(preference_alarms):
+            return None
+
+        if is_intrinsic_if(preference_alarms):
+            processed_alarms = copy.deepcopy(preference_alarms)
+            alarms_list = processed_alarms.get("Fn::If")
+            if not isinstance(alarms_list, list) or not len(alarms_list) == 3:
+                raise ValueError("Fn::If requires 3 arguments")
```

Comment:
nit: it looks like a generic check on `Fn::If`. I think it's better to refactor this part to intrinsics.py